tests/test_functools.py

```diff
@@ -119,47 +120,49 @@ def test_convert_kwargs_to_args():
 # ======================================================================================
 
 
-def test_allow_kwargs():
+def test_allow_only_kwargs():
     def f(a, /, b):
         # a is positional-only
         return a + b
 
     with pytest.raises(TypeError):
         f(a=1, b=2)
 
-    assert allow_kwargs(f)(a=1, b=2) == 3
+    assert allow_only_kwargs(f)(a=1, b=2) == 3
 
 
-def test_allow_kwargs_with_keyword_only_args():
+def test_allow_only_kwargs_with_keyword_only_args():
     def f(a, /, *, b):
         return a + b
 
     with pytest.raises(TypeError):
         f(a=1, b=2)
 
-    assert allow_kwargs(f)(a=1, b=2) == 3
+    assert allow_only_kwargs(f)(a=1, b=2) == 3
 
 
-def test_allow_kwargs_incorrect_number_of_args():
+def test_allow_only_kwargs_incorrect_number_of_args():
```

Comment:
Also for the rest, I would prefer "one test, one assert statement". But those are obvious enough that it is not worth to put more than 20min into it.